### ISSUE TITLE
Add user authentication

### ISSUE THREAD

**Issue by JakeWharton** [JakeWharton/u2020#82]
This allows

 * Grabbing the user photo for the nav drawer.
 * Starring and unstarring repos from the app.
 * Concrete demonstration of contextual debug actions.

**Comment by raskasa** [comment]
Do you have specific examples of contextual debug actions that you would want to show for a "Logged In" user?

cc @mattprecious

**Comment by JakeWharton** [comment]
The action is of logging in on the sign in screen itself.

**Comment by mattprecious** [comment]
Yes, this would be a mock-mode specific action that would just enter sane data to exercise the UI and get you through the screen with as few taps as possible.
